dicomanonymizer/simpledicomanonymizer.py

```diff
@@ -243,14 +243,68 @@ def anonymizeDICOMFile(inFile, outFile, extraAnonymizationRules = None):
     # Store modified image
     dataset.save_as(outFile)
 
+def get_private_tag(dataset, tag):
+    element = dataset.get(tag)
+
+    elementValue = element.value
+    tagGroup = element.tag.group
+    # The element is a private creator
+    if elementValue in dataset.private_creators(tagGroup):
+        creator = {
+            "tagGroup": tagGroup,
+            "creatorName": element.value
+        }
+        privateElement = None
+    # The element is a private element with an asociated private creator
+    else:
+        # Shift the element tag in order to get the creatorTag
+        # 0x1009 >> 8 will give 0x0010
+        creatorTagElement = element.tag.element >> 8
+        creatorTag = pydicom.tag.Tag(tagGroup, creatorTagElement)
+        creatorDataset = dataset.get(creatorTag)
+        creator = {
+            "tagGroup": tagGroup,
+            "creatorName": creatorDataset.value
+        }
+        # Define which offset should be applied to the creator to find
+        # this element
+        # 0x0010 << 8 will give 0x1000
+        offsetFromCreator = element.tag.element - (creatorTagElement << 8)
+        privateElement = {
+            "element": element,
+            "offset": offsetFromCreator
+        }
+
+    return {
+        "creator": creator,
+        "element": privateElement
+    }
+
 def anonymizeDataset(dataset, extraAnonymizationRules = None):
     currentAnonymizationActions = initializeActions()
 
     if extraAnonymizationRules != None:
         currentAnonymizationActions.update(extraAnonymizationRules)
 
+    privateTags = []
+
     for tag, action in currentAnonymizationActions.items():
         action(dataset, tag)
+        try:
+            element = dataset.get(tag)
+        except:
+            print("Cannot get element from tag: ", tag)
+
+        if element and element.tag.is_private:
+            privateTags.append(get_private_tag(dataset, tag))
 
     # X - Private tags = (0xgggg, 0xeeee) where 0xgggg is odd
     dataset.remove_private_tags()
+
+    # Adding back private tags if specified in dictionary
+    for privateTag in privateTags:
```

Comment:
what if the `action` on a private tag was to delete such tag ?

Reply:
Private tags are in the private_tags list only if the user defined a custom action. Else, the tags is deleted.

Reply:
> only if the user defined a custom action 
but maybe that action is a "delete" action.

Reply:
But that wouldn't make sense. Why creating a custom action to delete element if the element is already delete by default ?

Reply:
you give too much credit to the users :-) nonetheless such case is currently not supported (therefore bug) by the tool.

Said differently: you are ignoring the custom action optionally given by the user on private tags (there is more than keep or delete).

To me, private tags should be handled the same way normal tags are handled. It's just that their default action is "delete" but such action should be user customized like any other tag.

Reply:
Then, that's the case. I apply action to the private tags if it's listed in the dictionary.

Reply:
Understood, let's leave it like this for now then.

Reply:
To be clear, I apply the action to the tag before getting the data into the private_tags list. Then, I delete private tags and after, restore updated private tags